Userland/Libraries/LibGfx/DDSLoader.cpp

```diff
@@ -0,0 +1,1066 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Debug.h>
+#include <AK/Endian.h>
+#include <AK/LexicalPath.h>
+#include <AK/MappedFile.h>
+#include <AK/StringBuilder.h>
+#include <LibGfx/DDSLoader.h>
+#include <fcntl.h>
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#ifdef __serenity__
+#    include <serenity.h>
+#endif
+
+namespace Gfx {
+
+class Streamer {
+public:
+    Streamer(const u8* data, size_t size)
+        : m_data_ptr(data)
+        , m_size_remaining(size)
+    {
+    }
+
+    template<typename T>
+    bool read(T& value)
+    {
+        if (m_size_remaining < sizeof(T))
+            return false;
+        value = *((const LittleEndian<T>*)m_data_ptr);
```

Comment:
Please don't do this, it will end up doing misaligned reads.
Might want to extend AK/ByteReader.h to be a bit more generic, e.g. `ByteReader::load<T>(m_data_ptr)` (which would delegate to the appropriate `load(u8*, un&)` function and then return you a T with that.

Reply:
Swapped the reading to `InputMemoryStream`, so this shouldnt be a problem anymore?

Reply:
`InputMemoryStream` should technically be doing the right thing :tm:, so :+1: